python/lsst/sims/catUtils/utils/ObservationMetaDataGenerator.py

```diff
@@ -63,88 +120,26 @@ def __init__(self, database, driver='sqlite', host=None, port=None):
+        self._set_seeing_column(summary_columns)
 
-        if 'FWHMeff' in summary_columns:
-            self._seeing_column = 'FWHMeff'
-        else:
-            self._seeing_column = 'finSeeing'
-
-        # 27 January 2015
-        # self.columnMapping is an list of tuples.  Each tuple corresponds to a column in the opsim
-        # database's summary table.
-
-        # The 0th element of the tuple is how users will
-        # refer to the OpSim summary table columns (ie. how they are called in getObservationMetaDAta).
+        # Set up self.dtype containg the dtype of the recarray we expect back from the SQL query.
+        # Also setup baseQuery which is just the SELECT clause of the SQL query
         #
-        # The 1st element of the tuple is how the column is named in the OpSim db.
-        #
-        # The 2nd element of the tuple is how PhoSim refers to the quantity (as of OpSim3_61DBObject.py)
-        # (this is None if PhoSim does not expect the quantity)
-        #
-        # The 3rd element of the tuple is the datatype of the column
-        #
-        # The 4th element of the tuple is any coordinate transformation required to go from the user interface
-        # to the OpSim database (i.e. OpSim stores all angles in radians; we would like users to be
-        # able to specify angles in degrees)
-        #
-        # Note that this conforms to an older
-        # PhoSim API.  At some time in the future, both this and OpSim3_61DBObject.py
-        # (and possibly phoSimCatalogExamples.py) will need to be updated to
-        # reflect what PhoSim actually expects now.
-        #
-
-        # Just use the static method to assign this attribute
-        # to the mapping of names of phoSim to OpSim columns
-        self.columnMapping = self.OpSimColumnMap(self._seeing_column)
-
-        #Set up self.dtype containg the dtype of the recarray we expect back from the SQL query.
-        #Also setup baseQuery which is just the SELECT clause of the SQL query
-        #
-        #self.active_columns will be a list containing the subset of columnMapping columns
-        #that actually exist in this opsim database
+        # self.active_columns will be a list containing the subset of columnMapping columns
+        # that actually exist in this opsim database
         dtypeList = []
         self.baseQuery = 'SELECT'
         self.active_columns = []
-        for column in self.columnMapping:
-            if column[1] in summary_columns:
+        for column in self._user_interface_to_opsim:
+            rec = self._user_interface_to_opsim[column]
+            if rec[0] in summary_columns:
                 self.active_columns.append(column)
-                dtypeList.append((column[1],column[3]))
+                dtypeList.append((rec[0], rec[2]))
                 if self.baseQuery != 'SELECT':
                     self.baseQuery += ','
-                self.baseQuery += ' ' + column[1]
+                self.baseQuery += ' ' + rec[0]
```

Comment:
You are correct: I put this here so if people wanted to create their own "mock OpSim" cadence that did not contain all of the columns in the true OpSim summary table, the ```ObservationMetaDataGenerator``` would still work.

I think ignoring missing columns is the correct thing to do.  I don't want to try to calculating things (for example ```fiveSigmaDepth```) invisibly, in case users have their own models they want to use for missing values.